tests/unit/libutil/util.cc

```diff
@@ -99,7 +100,19 @@ TEST(base64Decode, decodeAString)
 
 TEST(base64Decode, decodeThrowsOnInvalidChar)
 {
-    ASSERT_THROW(base64Decode("cXVvZCBlcm_0IGRlbW9uc3RyYW5kdW0="), Error);
+    ASSERT_THROW(base64Decode("cXVvZCBlcm_0IGRlbW9uc3RyYW5kdW0="), FormatError);
+}
+
+TEST(base64Decode, decodeThrowsNoLeak)
+{
+    std::string msg = "cXVvZCBlcm_0IGRlbW9uc3RyYW5kdW0=";
+    try {
+        base64Decode(msg, true);
+    } catch (FormatError & e) {
+        EXPECT_THAT(e.msg(), testing::Not(testing::HasSubstr(msg)));
```

Comment:
```suggestion
        EXPECT_THAT(e.msg(), testing::Not(testing::HasSubstr(msg)));
        EXPECT_THAT(e.what(), testing::Not(testing::HasSubstr(msg)));
```

Reply:
This is still not future proof; please also add a functional test, which will cover the eventuality that yet another method is used for retrieving the message (e.g. something more structured than a string).